code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -45,7 +45,26 @@
 	if(is_pacified() && set_intent == I_HURT && !is_berserk())
 		to_chat(src, SPAN_WARNING("You don't want to harm other beings!"))
 		return
-	..()
+
+	. = ..()
+
+	// has the trait that lets mechanical eyes change their colour
+	if(HAS_TRAIT(src, TRAIT_DISABILITY_INTENT_EYES))
+		var/obj/item/organ/internal/eyes/eyes = get_eyes()
+		if(eyes && (eyes.status & ORGAN_ROBOT))
+			var/intent_colors = list(
```

Comment:
Don't recreate a list everytime this runs, make it a global list that everyone shares